### PR TITLE
EVM-373 overbroad permissioned files

### PR DESCRIPTION
**Pull request by yaziciahmet**
# Description

This PR addresses EVM-373, local files which are created within the application having too much permissions on themselves.

What is changed is that, all the permissions are given to only the owner and its group. The reasoning behind this decision is to allow the owners to have the flexibility to decide whether the files can be accessed from different users or not. If they want multiple users to access the files, they can put them into the same group, if not, they can put different users into different groups.

Affected files:
- genesis -> Read + Write -> genesis can be overwritten (without being deleted and re-created) with the `polygon-edge genesis` command, hence it should have write access as well
- default-config -> Read + Write -> same reason with genesis for the command `polygon-edge server export`
- snapshots and metadata -> Read + Write -> both are created from the function `writeDataStore`, and are updated on different runs (rather than being deleted and re-created)
- manifest -> Read + Write -> same reason with genesis for the command `polygon-edge manifest`
- private key and secrets -> Read only -> They are generated, and if they exist, they are never tried to be overridden, which allows them to be read only
- secret manager config -> Read + Write -> same reason with genesis for the command `polygon-edge secrets generate`
- node folders -> Read + Write + Exec -> The node directories are created with the function `createDir`. It should be ok to give all permissions to a folder, since it means the files can be seen and new files can be created within this directory.

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [x] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

Please complete this section if you ran manual tests for this functionality, otherwise delete it

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

**Review by Stefan-Ethernal — APPROVED** [review]
LGTM

**Review by b00ris — APPROVED** [review]

File: command/helper/helper.go [hunk 1]
```diff
@@ -238,8 +238,8 @@ func WriteGenesisConfigToDisk(genesisConfig *chain.Chain, genesisPath string) er
 	if err != nil {
 		return fmt.Errorf("failed to generate genesis: %w", err)
 	}
-
-	if err := os.WriteFile(genesisPath, data, os.ModePerm); err != nil {
+	/* #nosec */
```
Comment (JekaMas) [inline command/helper/helper.go:241]: why we need that comment?

File: command/helper/helper.go [hunk 2]
```diff
@@ -238,8 +238,8 @@ func WriteGenesisConfigToDisk(genesisConfig *chain.Chain, genesisPath string) er
...
+	if err := os.WriteFile(genesisPath, data, 0660); err != nil {
```
Comment (JekaMas) [inline command/helper/helper.go:242]: could we have one constant for that value in a separate package?

File: helper/common/common.go [hunk 3]
```diff
@@ -107,7 +107,7 @@ func createDir(path string) error {
 	}
 
 	if os.IsNotExist(err) {
-		if err := os.MkdirAll(path, os.ModePerm); err != nil {
+		if err := os.MkdirAll(path, 0770); err != nil {
```
Comment (JekaMas) [inline helper/common/common.go:110]: we still need 0770 here?

**Comment by Stefan-Ethernal** [comment]
Hey @JekaMas , I think it is better to review #1164, because that one is continuation of this PR. This would be eventually closed in favour of #1164.

**Pull request closed without merging**